protocol/directory_test.go

```diff
@@ -138,3 +139,231 @@ func TestRegisterExistedUserWithoutTB(t *testing.T) {
+		Username: "alice",
+		Key:      []byte("key")})
+	if err != Success {
+		t.Fatal("Unable to register")
+	}
+	// lookup in the same epoch
+	// expect a proof of absence
+	// and error ErrorNameNotFound
+	res, err := d.KeyLookup(&KeyLookupRequest{Username: "bob"})
+	if err != ErrorNameNotFound ||
+		res.Error != ErrorNameNotFound {
+		t.Fatal("Expect error code", ErrorNameNotFound, "got", err)
+	}
+	if ap := res.AP; ap == nil || bytes.Equal(ap.LookupIndex, ap.Leaf.Index) {
+		t.Fatal("Expect a proof of absence")
+	}
+	if res.TB != nil {
+		t.Fatal("Expect returned TB is nil")
+	}
+
+	d.Update()
+	// lookup in epoch after registering epoch
+	// expect a proof of absence
+	// and error ErrorNameNotFound
+	res, err = d.KeyLookup(&KeyLookupRequest{Username: "bob"})
+	if err != ErrorNameNotFound ||
+		res.Error != ErrorNameNotFound {
+		t.Fatal("Expect error code", ErrorNameNotFound, "got", err)
+	}
+	if ap := res.AP; ap == nil || bytes.Equal(ap.LookupIndex, ap.Leaf.Index) {
+		t.Fatal("Expect a proof of absence")
+	}
+	if res.TB != nil {
+		t.Fatal("Expect returned TB is nil")
+	}
+}
+
+func TestKeyLookupWithoutTB(t *testing.T) {
+	d := newDirectory(t, false)
+	_, err := d.Register(&RegistrationRequest{
+		Username: "alice",
+		Key:      []byte("key")})
+	if err != Success {
+		t.Fatal("Unable to register")
+	}
+	// lookup in the same epoch
+	// expect a proof of absence
+	// and error ErrorNameNotFound
+	res, err := d.KeyLookup(&KeyLookupRequest{Username: "alice"})
+	if err != ErrorNameNotFound ||
+		res.Error != ErrorNameNotFound {
+		t.Fatal("Expect error code", ErrorNameNotFound, "got", err)
+	}
+	if ap := res.AP; ap == nil || bytes.Equal(ap.LookupIndex, ap.Leaf.Index) {
+		t.Fatal("Expect a proof of absence")
+	}
+	if res.TB != nil {
+		t.Fatal("Expect returned TB is nil")
+	}
+
+	d.Update()
+	// lookup in epoch after registering epoch
+	// expect a proof of inclusion
+	res, err = d.KeyLookup(&KeyLookupRequest{Username: "alice"})
+	if ap := res.AP; ap == nil || !bytes.Equal(ap.LookupIndex, ap.Leaf.Index) {
+		t.Fatal("Expect a proof of inclusion")
+	}
+	if res.TB != nil {
+		t.Fatal("Expect returned TB is nil")
+	}
+}
+
+func TestKeyLookupNameNotFoundWithoutTB(t *testing.T) {
+	d := newDirectory(t, false)
+	_, err := d.Register(&RegistrationRequest{
+		Username: "alice",
+		Key:      []byte("key")})
+	if err != Success {
+		t.Fatal("Unable to register")
+	}
+	// lookup in the same epoch
+	// expect a proof of absence
+	res, err := d.KeyLookup(&KeyLookupRequest{Username: "alice"})
+	if err != ErrorNameNotFound ||
+		res.Error != ErrorNameNotFound {
+		t.Fatal("Expect error code", ErrorNameNotFound, "got", err)
+	}
+	if ap := res.AP; ap == nil || bytes.Equal(ap.LookupIndex, ap.Leaf.Index) {
+		t.Fatal("Expect a proof of absence")
+	}
+	if res.TB != nil {
+		t.Fatal("Expect returned TB is nil")
+	}
+
+	d.Update()
+	// lookup in different epochs
+	// expect a proof of absence
+	res, err = d.KeyLookup(&KeyLookupRequest{Username: "bob"})
```

Comment:
This part of the test seems redundant to me since it's no different than in the analogous `WithTB` test. But maybe that's exactly what you're testing?